### DISCUSSION TITLE
How to get non null embedded resources ?

### DISCUSSION THREAD

**Discussion by Spiritus44** [PostgREST/postgrest#1747] [Q&A] [answered]
Hello,

I'm trying to understand how can I get all the none null embedded resources.

I have made a database that almost the same as the example in the documentation, so I will use the documentation schema to explain to you what I tried to do.

In [this] (https://postgrest.org/en/v4.1/_images/film.png) schema I would like to know how can I get all the Films that have a Roles (all the Film that are pointed by a Roles) and do not return the others.

I tryed something like that:
`GET /films?select=*,roles(*)&roles.character=not.is.null HTTP / 1.1`

But it doesn't works: it returns all the films even the one where no Roles are defines.

Do someone can help me ?

Thank you for your time and your answers.

**Comment by wolfgangwalther** [comment, marked as answer]
Currently, it's not possible to filter the main query on the embedded columns. This is asked for in #1075.

For now, you could add a view that joins those two tables with an inner join and then query this.
